app/serializers/rest/application_serializer.rb

```diff
@@ -1,24 +1,18 @@
 # frozen_string_literal: true
 
 class REST::ApplicationSerializer < ActiveModel::Serializer
-  attributes :id, :name, :website, :scopes, :redirect_uri,
-             :client_id, :client_secret
+  attributes :id, :name, :website, :scopes, :redirect_uris
 
   # NOTE: Deprecated in 4.3.0, needs to be removed in 5.0.0
   attribute :vapid_key
 
+  # We should consider this property deprecated for 4.3.0
+  attribute :redirect_uri
+
   def id
     object.id.to_s
   end
 
-  def client_id
-    object.uid
-  end
-
-  def client_secret
-    object.secret
-  end
```

Comment:
In 4.2.x, the properties of `Application` for `client_id` and `client_secret` were optional; it's only in `main` that `client_id` started to be always returned.

Reply:
Calling them optional is confusing.

If my understanding is correct:
- before #27142, `client_id` and `client_secret` were only present in the response to `POST /api/v1/apps`
- after #27142, `client_secret` is still only present in the response to `POST /api/v1/apps` but `client_id` is exposed in response to `GET /api/v1/apps/verify_credentials` as well

And your change would revert to before #27142.

Reply:
yup, which I think is fine & intentional, since #27142 was never released. This makes scopes and redirect_uris always returned inline with that PR. I don't know if you need `client_id` in verify?

Reply:
we can moved it (`client_id`) back in I guess? I can't recall why I added it to the verify response